Classes/Hooks/CmdMapLocalizationHook.php

```diff
@@ -0,0 +1,207 @@
+                if (
+                    isset($tableTCAColumns[$field])
+                    && isset($tableTCAColumns[$field]['config']['type'])
+                    && $tableTCAColumns[$field]['config']['type'] === 'slug'
+                ) {
+                    $detectedSlugField = $field;
+                    continue;
+                }
+                if (
+                    !isset($tableTCAColumns[$field])
+                    || !isset($tableTCAColumns[$field]['l10n_mode'])
+                    || $tableTCAColumns[$field]['l10n_mode'] !== 'deepl'
+                ) {
+                    continue;
+                }
+                $translatedContent = $deepLService->translateRequest(
+                    $originalElement[$field],
+                    $targetLanguage['language_isocode'],
+                    $sourceLanguage['language_isocode']
+                );
+                if (!empty($translatedContent) && isset($translatedContent['translations'])) {
+                    foreach ($translatedContent['translations'] as $translation) {
+                        if ($translation['text'] != '') {
+                            $value = $translation['text'] ?? $value;
+                            $deepLTranslated = true;
+                            break;
+                        }
+                    }
+                }
+                $translatedElement[$field] = $value ?? $originalElement[$field];
+            }
+            if ($deepLTranslated) {
+                // empty slug field to auto create
+                if ($detectedSlugField !== '') {
+                    $translatedElement[$detectedSlugField] = '';
+                }
+                $data[$table][$translatedElement['uid']] = $translatedElement;
+                $innerDataHandler = GeneralUtility::makeInstance(DataHandler::class);
+                $innerDataHandler->start($data, []);
+                $innerDataHandler->process_datamap();
+                GeneralUtility::makeInstance(PageRepository::class)
+                    ->markPageAsTranslatedWithDeepl($siteInformation['pageUid'], $targetLanguage);
+            }
+        }
+    }
+
+    private function cleanUpWithDefault(
+        string $table,
+        array $originalElement,
+        array $translatedElement,
+        DataHandler $dataHandler,
+        array $tableTCA,
+        array $targetLanguage
+    ): void {
+        // load some defaults
+        $pageId = $table === 'pages' ? $originalElement['uid'] : $originalElement['pid'];
+        $TSConfig = BackendUtility::getPagesTSconfig($pageId)['TCEMAIN.'] ?? [];
+        $tableEntries = $dataHandler->getTableEntries($table, $TSConfig);
+        if (!empty($TSConfig['translateToMessage']) && !($tableEntries['disablePrependAtCopy'] ?? false)) {
+            $translateToMsg = $this->getLanguageService()->sL($TSConfig['translateToMessage']);
+            $translateToMsg = @sprintf($translateToMsg, $targetLanguage['title']);
```

Comment:
Error suppressing with `@` is bad habit. This does not solve an issue, it silences/hides the issue away. 

What is the reason you used it ?

Reply:
This part is copied 1:1 from DataHandler, see https://github.com/TYPO3-CMS/core/blob/main/Classes/DataHandling/DataHandler.php#L4765
Should avoid an sprintf to empty string or no markers et, I guess.

Reply:
Still bad habit - albit not yours ;). Okay.